### PR TITLE
feat: Sentry Browser Error tracking / React, Solid, Vue, Plain HTML

### PR DESCRIPTION
**Pull request by aheissenberger**
Add a support for error tracking with sentry in React Browser Apps.

* any errors in the browser are captured and forwarded to sentry.io
* sourcemaps are uploaded to sentry.io
* errors and source map are connected to releases (git commit hash)
* no deep integration with the Vike client side router (help needed!)

See `README.md` for information on setup.

**Example:**
`pnpm cli --react --sentry`

### CODE REVIEW

**Comment by aheissenberger** [comment]
@magne4000 after this being merged I will provide pull requests for `vue` and `solid` and pure `javascript`

**Comment by magne4000** [comment]
FYI I expect [vue](https://docs.sentry.io/platforms/javascript/guides/vue/) and [solid](https://docs.sentry.io/platforms/javascript/guides/solid/) to have the same level of features as `react` when possible.
So if you feel like doing those, go ahead. If not, please close this PR and open a new one targeting a branch other than `main`, so that I can add those integration before releasing it.

**Comment by magne4000** [comment]
> @magne4000 after this being merged I will provide pull requests for `vue` and `solid` and pure `javascript`

**Comment by aheissenberger** [comment]
> FYI I expect [vue](https://docs.sentry.io/platforms/javascript/guides/vue/) and [solid](https://docs.sentry.io/platforms/javascript/guides/solid/) to have the same level of features as `react` when possible. So if you feel like doing those, go ahead. If not, please close this PR and open a new one targeting a branch other than `main`, so that I can add those integration before releasing it.
> 
> EDIT: Just saw your comment :). If possible please create another branch for all of this then

OK - I will add vue, solid and plain javascript to this pull request

**Comment by aheissenberger** [comment]
@magne4000 everything except `vue` are ready. Can you help with this?
Based on the documentation from sentry I will need access to the `Vue` instance which is hidden inside `vike-vue`.

I tried this https://github.com/aheissenberger/bati/blob/aeacdd61a2b5f31e12fe8d92d1ad852f11f8ff87/boilerplates/sentry/files/sentry.browser.config.ts#L14-L30 but I still get this error:
```
errorhandler.js:6 Uncaught TypeError: Cannot destructure property 'errorHandler' of 'e.config' as it is undefined.
    at by (errorhandler.js:6:11)
    at mi (integration.js:74:3)
    at xy (integration.js:50:5)
    at Object.setup (integration.js:25:7)
    at Yi (integration.js:117:17)
    at integration.js:82:7
    at Array.forEach (<anonymous>)
    at _f (integration.js:79:16)
    at fp._setupIntegrations (baseclient.js:394:26)
    at fp.init (baseclient.js:259:12)
    at bf (sdk.js:34:10)
    at jh (sdk.js:161:18)
    at Ay (sdk.js:27:10)
    at sentry.browser.config.ts:7:3
```

Based on [Sentry Vue Docs](https://docs.sentry.io/platforms/javascript/guides/vue/) I need for `Vue 3` the `app` instance!

**Comment by magne4000** [comment]
@aheissenberger I think that you need to initialize it in [onCreateApp](https://github.com/vikejs/vike-vue/blob/3891a6f3d0826f69d7b0bacacbe01c31f82c6ca7/examples/full/pages/%2BonCreateApp.ts#L7) hook.

**Comment by aheissenberger** [comment]
> @aheissenberger I think that you need to initialize it in [onCreateApp](https://github.com/vikejs/vike-vue/blob/3891a6f3d0826f69d7b0bacacbe01c31f82c6ca7/examples/full/pages/%2BonCreateApp.ts#L7) hook.

I got it working 😄 

I added it to `boilerplates/vue/files/layouts/LayoutDefault.vue`:
https://github.com/aheissenberger/bati/blob/07ab567dea7d595072597e0c75f53c8bc63c9909/boilerplates/vue/files/layouts/LayoutDefault.vue#L19-L20

I found out, that the `app` can be accessed in the `PageContext`. I tried `pages/+client.ts` but failed - what worked was `LayoutDefault.vue`.

If you know a better place please suggest a change.

File: boilerplates/sentry/files/$vite.config.ts.ts
Comment (magne4000) [inline boilerplates/sentry/files/$vite.config.ts.ts:20]: Is this necessary? Vite already loads `.env` files https://vitejs.dev/guide/env-and-mode#env-files

File: boilerplates/sentry/files/$vite.config.ts.ts
Comment (magne4000) [inline boilerplates/sentry/files/$vite.config.ts.ts:25]: What does this do?

File: boilerplates/sentry/files/$vite.config.ts.ts
Comment (magne4000) [inline boilerplates/sentry/files/$vite.config.ts.ts:34]: We could probably use `import.meta.env` instead

File: boilerplates/sentry/files/pages/sentry/+Page.tsx
Comment (magne4000) [inline boilerplates/sentry/files/pages/sentry/+Page.tsx:1]: Please create 3 new boilerplates `react-sentry`, `vue-sentry` and `solid-sentry` for this purpose (With the right `flag` combination in package.json).

This file should then be split and moved to its respective boilerplates.

File: boilerplates/sentry/files/pages/sentry/+Page.js
Comment (magne4000) [inline boilerplates/sentry/files/pages/sentry/+Page.js:9]: Those `import.meta.env` are interpreted during Bati CLI execution, so here this probably always return `DEV`. Do not bother too much with pure HTML though, just make it as simple as possible.

File: boilerplates/sentry/files/pages/sentry/+Page.vue
Comment (magne4000) [inline boilerplates/sentry/files/pages/sentry/+Page.vue:1]: Move into `vue-sentry`

File: boilerplates/sentry/files/sentry.browser.config.ts
Comment (magne4000) [inline boilerplates/sentry/files/sentry.browser.config.ts:1]: Split and move to their respective boilerplates

File: boilerplates/vue/files/layouts/LayoutDefault.vue
Comment (magne4000) [inline boilerplates/vue/files/layouts/LayoutDefault.vue:23]: Prefer this syntax when you can. Unused imports are automatically removed.

```suggestion
import { sentryBrowserConfig } from "../sentry.browser.config";
//# BATI.has("sentry")
sentryBrowserConfig();
```

File: packages/cli/rules.ts
Comment (magne4000) [inline packages/cli/rules.ts:41]: Seems unnecessary, as this is always true. You can just remove the rule entirely.

**Review by magne4000 — CHANGES_REQUESTED** [review]

File: boilerplates/sentry/files/$vite.config.ts.ts
Comment (aheissenberger) [reply boilerplates/sentry/files/$vite.config.ts.ts:24]: yes, but to access `.env` in `vite.config.ts` we would have to wrap `defineConfig` to access the loaded environment variables:
https://stackoverflow.com/questions/66389043/how-can-i-use-vite-env-variables-in-vite-config-js

If we go down this route, `magicast` will break as currently there is no helper to add Vite plugins or other stuff if wrapped 😞 . I have looked into this and magi cast deeply but did not find any solution except to import `dotenv`.

File: boilerplates/sentry/files/$vite.config.ts.ts
Comment (aheissenberger) [reply boilerplates/sentry/files/$vite.config.ts.ts:30]: This will create the function call to activate the `dotenv` library and add it to the file:
`config()`

Comment (magne4000) [reply boilerplates/sentry/files/$vite.config.ts.ts:30]: It should probably be specified in a comment then

File: boilerplates/sentry/files/$vite.config.ts.ts
Comment (aheissenberger) [reply boilerplates/sentry/files/$vite.config.ts.ts:39]: if we use `import.meta.env` we would have to prefix the names with `PUBLIC_` which would leak `SENTRY_AUTH_TOKEN` to the APP. 

I placed all this values in `.env` to support the follow use cases:
* Sentry Browser which only need `PUBLIC_ENV__SENTRY_DSN`
* SentryVitePlugin which does the adding of the debug id, release and uploads the source map files - needs the rest (this needs to be done for the browser code and the server code if Sentry Server is used)
* Sentry Server (e.g. Sentry for AWS) uses `SENTRY_DSN` which can be the same as `PUBLIC_ENV__SENTRY_DSN` but this is optional.

I am too not happy to include `dotenv` but did not found any other way except manual loading `.env` and parsing. I open to any solution which removes the need to use `dotenv`

File: boilerplates/sentry/files/pages/sentry/+Page.js
Comment (aheissenberger) [reply boilerplates/sentry/files/pages/sentry/+Page.js:9]: No, `import.meta.env` is different depending on how the code is run `pnpm dev` or `pnpm build && pnpm preview`. This page should help to find problems with the `Sentry Integration`.
The current setup does not include sentry in DEV and if you would try to test the integration it would fail.

File: boilerplates/sentry/files/pages/sentry/+Page.vue
Comment (aheissenberger) [reply boilerplates/sentry/files/pages/sentry/+Page.vue:1]: I have tried to keep everything in one place to make global changes easier but will follow your advices 😄

File: boilerplates/sentry/files/pages/sentry/+Page.js
Comment (magne4000) [reply boilerplates/sentry/files/pages/sentry/+Page.js:9]: Oh, I though this was a `$+Page.js.ts` file, nevermind

**Comment by aheissenberger** [comment]
finished:
* split to multiple boilerplates
* remove of `dotenv`

**Comment by magne4000** [comment]
Thank you for this integration! Good to merge once the CI is finished.

**Pull request merged**
